### ISSUE TITLE
Flameshot will not paste into Kami on Chrome >91 in Ubuntu/Pop (20.04/21.04) but works on Fedora/Manjaro

### ISSUE THREAD

**Issue by spxak1** [flameshot-org/flameshot#1864]
**Flameshot version**
Flameshot v0.10.1
Compiled with Qt 5.12.8

**Describe the bug**
Pasting images on Kami running on Chrome version greater than 91 does not work.
You can paste on Google Docs on Chrome, and on Gimp, but not on Kami.
If you delete the user files ```.config/google-chrome``` and start again, it will work for the first launch, but not after.
Even if you only open Chrome once, and close it, after you delete the user files, it will not paste the second time.
So far the issue is only with Kami, others report gitlab, but I have not tested.
shift+ctrl+prntScreen works fine.

See video here: https://streamable.com/4cksfd

**To Reproduce**
* Install PopOS or Ubuntu 21.04
* Install Chrome version greater than 91
* Install flameshot (from apt from git), either version 0.8 or 10.
* Launch Chrome connect to Kami, start a document
* Take a screenshot with ```flameshot gui```
* Paste on Kami. It works
* Close Chrome.
* Repeat launching Chrome, connect to kami, take a screenshot. Nothing pastes!
* Open Google docs, paste works fine.
* Install gimp
* Launch gimp, create a new file. flameshot pastes fine
* Try to paste on Chrome/Kami, nothing pastes.
* Delete ```./config/google-chrome```
* Start Chrome again and Kami, paste works again.
* Close Chrome and start, paste does not work again.

**Expected behavior**
Screen shot pastes on Kami in Chrome versions greater than 91 as they do in version 91.

**System Information**
Pop_OS 21.04 but also tested on Ubuntu 21.04. On four different computers, with new installs, new users.

**Comment by spxak1** [comment]
Additional info:

* Flameshot "Use JPG format for clipboad" makes no difference.

* ```xclip -o -t TARGETS -selection clipboard``` when using flameshot gives
~~~
application/x-qt-image
image/png
image/bmp
image/cur
image/ico
image/jpeg
image/jpg
image/pbm
BITMAP
image/pgm
image/ppm
PIXMAP
image/xbm
image/xpm
TARGETS
MULTIPLE
TIMESTAMP
SAVE_TARGETS
~~~

* when using ctrl+shift+prn gives
~~~
image/png
TARGETS
TIMESTAMP
~~~

Again, both paste fine on Gimp and Chrome with Google Docs.

**Comment by spxak1** [comment]
Also interestingly enough (and this just happened now), if I use ctrl+shift+prn and then ctrl+v in this box,

it does this (I have actually only pressed ctrl+v for this): 

![image](https://user-images.githubusercontent.com/29977030/131968554-01b5d7c5-b5f5-4dc1-9ff4-f662518992a9.png)


But if I use flameshot, ctrl+v does nothing.

**Comment by spxak1** [comment]
No need to try Kami. 

The problem occurs right here on github too!!!

Images paste fine when acquired with ctrl+shift+prn, but nothing with flameshot. 

If I clear user files, the first time Chrome launches, it works again, stops after that.

**Comment by mmahmoudian** [comment]
@spxak1 You repeatedly mention "ctrl+shift+prn" but I don't know what that keybinding is bound to. Would you please dig out that information that what is running when you press "ctrl+shift+prn" ?

**Comment by spxak1** [comment]
> @spxak1 You repeatedly mention "ctrl+shift+prn" but I don't know what that keybinding is bound to. Would you please dig out that information that what command is running when you press "ctrl+shift+prn" ?

Sorry, that's gnome-screenshot.

> I tried with Ubuntu 18.04 and Flameshot v0.10.1 from Snap

I have replicated the issue in 5 systems. Clean 21.04 install, flameshot from apt or github, Chrome 91 works fine, anything later (92 or greater won't). 

> As far as my logic goes, if pasting works the first time, the data in the clipboard is fine and if closing and opening the browser changes the situation, either the content of clipboard has changed or something else. Considering that the Flameshot has not been executed anytime in-between, I don't see how Flameshot can be the faulty one in the process.

I can see the point there. Please see if you can replicate the issue to confirm. 

Thank you for your time looking into it.

**Comment by unwelt** [comment]
I've encountered this problem too. Ubuntu 20.04, Google Chrome Version 93.0.4577.63. When I'm taking a screenshot by Flameshot I can't paste a result on, for example, Github. I pressed ctr+v and nothing happened. But, when I'm trying to paste a result to social network vk.com, I see an error in Chrome Developer Tools. Maybe It will help you to catch the problem
![image](https://user-images.githubusercontent.com/16933459/133352360-e9d63515-2d3b-436b-b08b-28c7a15065dc.png)

**Comment by alexfornuto** [comment]
[This solution](https://github.com/flameshot-org/flameshot/issues/1840#issuecomment-916774959) worked for me.

**Comment by unwelt** [comment]
> [This solution](https://github.com/flameshot-org/flameshot/issues/1840#issuecomment-916774959) worked for me.

Thank you! It worked for me too.

**Comment by spxak1** [comment]
> [This solution](https://github.com/flameshot-org/flameshot/issues/1840#issuecomment-916774959) worked for me.

Thank you for the suggestion. How do I go about disabling Ozone? Sorry for the naive question.

**Comment by altuhovsu** [comment]
Strange, for me Copy/Paste here is not working (from Flameshot directly).

But copy/paste works "via Krita (KDE image editor)"
1. Create screenshot usind Flameshot and save to clipboard
2. Paste in the empty/new file in the Krita = Ctrl+V works!
3. Press Ctrl+C in the Krita = works
4. Press Ctrl+V here (in the Github comment) = works (image uploaded)

Kubuntu 21.10 KDE 5.23.2 on Wayland!!
Chrome  95.0.4638.69 (Official Build) (64-bit)
Flameshot v0.10.1 Compiled with Qt 5.12.8 = deb x64 for 20.04

**Comment by p-himik** [comment]
GitHub works for me but GitLab fails with this error in the JS console:
```
Uncaught TypeError: Failed to execute 'append' on 'FormData': parameter 2 is not of type 'Blob'.
    at A (commons-pages.admin.topics.edit-pages.admin.topics.new-pages.groups.milestones.edit-pages.groups.mil-8657eee6.b7f355f7.chunk.js:1)
    at z (commons-pages.admin.topics.edit-pages.admin.topics.new-pages.groups.milestones.edit-pages.groups.mil-8657eee6.b7f355f7.chunk.js:1)
    at HTMLTextAreaElement.<anonymous> (commons-pages.admin.topics.edit-pages.admin.topics.new-pages.groups.milestones.edit-pages.groups.mil-8657eee6.b7f355f7.chunk.js:1)
    at HTMLTextAreaElement.dispatch (main.0fbb4b89.chunk.js:44)
    at HTMLTextAreaElement.g.handle (main.0fbb4b89.chunk.js:44)
```

Looking at the code, the issue it due to `image.getAsFile()` returning `null`.
That `image` is an instance of `DataTransferStream` with `kind` and `type` set to values that depend on the application used to copy the image.
I did some experiments and here's the result:
- Gwenview: `kind: "file", type: "image/jpeg"` - everything works on GitLab
- Gimp: `kind: "string", type: "image/bmp" - doesn't work
- Flameshot (via PNG): `kind: "string", type: "application/x-qt-image"` - doesn't work
- Flameshot (via JPG): `kind: "string", type: "image/jpeg" - doesn't work

When `kind` is not `"file"`, `getAsFile()` returns `null`, and it's a documented behavior.

Looking at the originating `ClipboardEvent`, I can see this:
![image](https://user-images.githubusercontent.com/4410314/141346518-5c008057-2a2c-4b06-afad-28ec02109abb.png)
A very similar picture can be seen when using Gimp. But when using Gwenview, there's only one entry - the one in the list above.

And the image pasting code that GitLab uses just picks up the very first item with the word "image" in its `type`. Obviously, that's completely wrong. But I also can't understand why all but one of the apparent images that both Flameshot and Gimp put in the clipboard have `kind: "string"`.

**Comment by mmahmoudian** [comment]
@p-himik thanks for digging into this deeper. Not it is a bit more clear (at least to me) that what the issue stems from. 

@veracioux & @borgmanJeremy  is there any downside of creating image file (the kind and type) and then push it clipboard?

**Comment by veracioux** [comment]
@mmahmoudian I don't think we would gain anything from that. We'd still have to read the image file into a byte array and push it to the clipboard, which is what we are doing now.
